client/components/signup-site-title/index.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import LoggedOutForm from 'components/logged-out-form';
+
+import formState from 'lib/form-state';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+
+import { getSiteTitle } from 'state/signup/steps/site-title/selectors';
+import { setSiteTitle } from 'state/signup/steps/site-title/actions';
+
+const SignupSiteTitle = React.createClass( {
+	displayName: 'Signup Site Title',
+
+	contextTypes: {
+		store: React.PropTypes.object,
+	},
+
+	propTypes: {
+		onSubmit: React.PropTypes.func.isRequired,
+	},
+
+	getInitialState() {
+		return {
+			form: {
+				siteTitle: {
+					value: this.props.siteTitle
+				}
+			}
+		};
+	},
+
+	componentWillMount() {
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'siteTitle' ],
+			validatorFunction: () => {},
+			onNewState: this.setFormState,
+			onError: this.handleFormControllerError,
+			hideFieldErrorsOnChange: true,
+			initialState: this.state.form
```

Comment:
There's some indirection here, in that `initialState` comes from `this.state.form`, which itself is derived from `this.props.siteTitle`. Could we ditch the `getInitialState` method, just refer to `this.props.siteTitle` in the controller's `initialState` option, and let the controller be the only piece setting component state (using `this.setFormState`)? (If not, that's fine)